lib/node_modules/@stdlib/array/base/cuany-by-right/lib/assign.js

```diff
@@ -0,0 +1,291 @@
+
+	flg = false;
+	io = offset + (x.length - 1) * stride;
+	for ( i = x.length - 1; i >= 0; i-- ) {
+		if ( flg === false && x[ i ] ) {
+			flg = true;
+		}
+		y[ io ] = flg;
+		io -= stride;
+	}
+	return y;
+}
+
+/**
+* Cumulatively tests whether at least one element in a provided accessor array is truthy, iterating from right to left.
+*
+* @private
+* @param {Object} x - input array object
+* @param {Object} y - output array object
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @returns {Collection} output array
+*
+* @example
+* var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
+* var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+*
+* var x = toAccessorArray( [ false, false, true, false, false ] );
+* var y = toAccessorArray( [ false, null, false, null, false, null, false, null, false, null ] );
+*
+* var arr = accessors( arraylike2object( x ), arraylike2object( y ), 2, 0 );
+*
+* var v = y.get( 0 );
+* // returns false
+*
+* v = y.get( 2 );
+* // returns false
+*
+* v = y.get( 4 );
+* // returns true
+*
+* v = y.get( 6 );
+* // returns true
+*
+* v = y.get( 8 );
+* // returns true
+*/
+function accessors( x, y, stride, offset ) {
+	var xdata;
+	var ydata;
+	var xget;
+	var yset;
+	var flg;
+	var io;
+	var i;
+
+	xdata = x.data;
+	ydata = y.data;
+
+	xget = x.accessors[ 0 ];
+	yset = y.accessors[ 1 ];
+
+	flg = false;
+	io = offset + (xdata.length - 1) * stride;
+	for ( i = xdata.length - 1; i >= 0; i-- ) {
+		if ( flg === false && xget( xdata, i ) ) {
+			flg = true;
+		}
+		yset( ydata, io, flg );
+		io -= stride;
+	}
+	return ydata;
+}
+
+/**
+* Cumulatively tests whether at least one element in a provided complex number array is truthy, iterating from right to left.
+*
+* @private
+* @param {Collection} x - array containing interleaved real and imaginary components
+* @param {Object} y - output array object
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @returns {Collection} output array
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+* var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
+* var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+*
+* var x = new Float64Array( [ 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0 ] );
+* var y = toAccessorArray( [ false, null, false, null, false, null, false, null, false, null ] );
+*
+* var arr = complex( x, arraylike2object( y ), 2, 0 );
+*
+* var v = y.get( 0 );
+* // returns false
+*
+* v = y.get( 2 );
+* // returns false
+*
+* v = y.get( 4 );
+* // returns true
+*
+* v = y.get( 6 );
+* // returns true
+*
+* v = y.get( 8 );
+* // returns true
+*/
+function complex( x, y, stride, offset ) {
+	var ydata;
+	var yset;
+	var flg;
+	var io;
+	var i;
+
+	yset = y.accessors[ 1 ];
+	ydata = y.data;
+
+	flg = false;
+	io = offset + (x.length/2 - 1) * stride;
```

Comment:
Should this be `*2` rather than `/2`?

Reply:
Dividing by 2 is necessary because each complex number is represented by 2 elements. Without this division, you would not correctly determine the position of the complex number pairs